### PR TITLE
Lint did not recognize multiple imports of ApiModules

### PR DESCRIPTION
**Pull request by rshunim**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6196

## Description
Lint does not recognize multiple imports of ApiModules in an integration, as a result, lint would only copy the first / one of the ApiModule imports.

### CODE REVIEW

File: demisto_sdk/commands/lint/helpers.py [hunk 1]
```diff
@@ -280,22 +280,21 @@ def add_tmp_lint_files(
...
-                    else:
-                        url = f"https://raw.githubusercontent.com/demisto/content/master/{rel_api_path}"
-                        api_content = requests.get(url=url, verify=False).content
-                        cur_path.write_bytes(api_content)
+                if module_match := re.findall(module_regex, data):
+                    for module_name in module_match:
+                        rel_api_path = (
+                            Path("Packs/ApiModules/Scripts")
+                            / module_name
+                            / f"{module_name}.py"
```
Comment (dorschw) [inline demisto_sdk/commands/lint/helpers.py:288]:
```suggestion
                            CONTENT_PATH /
                            "Packs/ApiModules/Scripts") /
                            f"module_name/{module_name}.py"
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## Unreleased
 * Added the pack version to the code files when calling **unify**. The same value is removed when calling **split**.
 * Contribution PRs that update outdated packs now display a warning message.
+* Fixed an issue where **lint** did not recognize multiple imports of ApiModules.
```
Comment (dorschw) [inline CHANGELOG.md:6]:
```suggestion
* Fixed an issue where **lint** would not properly handle multiple ApiModules imports.
```

File: demisto_sdk/commands/lint/helpers.py [hunk 3]
```diff
@@ -280,22 +280,21 @@ def add_tmp_lint_files(
...
+                        )
+                        cur_path = pack_path / f"{module_name}.py"
```
Comment (dorschw) [inline demisto_sdk/commands/lint/helpers.py:290]: I'd use `api_module_code_path`, as `cur_path` doesn't tell future maintainers where this path leads

File: demisto_sdk/commands/lint/helpers.py [hunk 4]
```diff
@@ -280,22 +280,21 @@ def add_tmp_lint_files(
...
+                        if content_repo:
```
Comment (dorschw) [inline demisto_sdk/commands/lint/helpers.py:291]: is `content_repo` optional? The type hints say it's mandatory, and should not be `None`. 

If it's optional, please fix the type hints.
If it's mandatory, please explain this (line 291) condition :)

File: demisto_sdk/commands/lint/helpers.py [hunk 5]
```diff
@@ -280,22 +280,21 @@ def add_tmp_lint_files(
...
+                            module_path = content_repo / rel_api_path
+                            shutil.copy(src=module_path, dst=cur_path)
+                        else:
+                            url = f"https://raw.githubusercontent.com/demisto/content/master/{rel_api_path}"
+                            api_content = requests.get(url=url, verify=False).content
+                            cur_path.write_bytes(api_content)
```
Comment (dorschw) [inline demisto_sdk/commands/lint/helpers.py:297]: Please use `get_remote_file` here if possible. If not, ping me

**Review by dorschw — COMMENTED** [review]
Nice. Do/can we unit-test this behavior?

File: demisto_sdk/commands/lint/tests/helper_test.py [hunk 6]
```diff
@@ -253,6 +257,28 @@ def test_split_warnings_errors(
     assert other == output_other
 
 
+def test_add_tmp_lint_files(mocker, repo):
```
Comment (dorschw) [inline demisto_sdk/commands/lint/tests/helper_test.py:260]: please add a `given when then` docstring

File: demisto_sdk/commands/lint/tests/helper_test.py [hunk 7]
```diff
@@ -253,6 +257,28 @@ def test_split_warnings_errors(
...
+    pack.create_script(code="# TEST2ApiModule")
+    integration_code = "from TEST1ApiModule import *\nfrom TEST2ApiModule import *"
+    integration = pack.create_integration(name="test", code=integration_code)
+
+    mock_copy = mocker.patch.object(shutil, "copy", return_value=None)
+    content_repo = Path(repo.path)
+    pack_path = Path(pack.path)
+    lint_files = [Path(integration.code.path)]
+
+    with add_tmp_lint_files(
```
Comment (dorschw) [inline demisto_sdk/commands/lint/tests/helper_test.py:272]: `with` - is this context? 🤔

**Comment by rshunim** [comment]
Waiting for the [nightly](https://code.pan.run/xsoar/content/-/pipelines/5052133)

**Review by dorschw — APPROVED** [review]

**Pull request merged**